packages/contracts/test/SphinxUtils.t.sol

```diff
@@ -14,6 +15,154 @@ import {
 contract SphinxUtils_Test is Test, SphinxUtils {
     function setUp() public {}
 
+    function test_checkFork_passes() external {
+        assertEq(checkFork(), true);
+    }
+
+    function test_sphinxCheckAccesses_fails_did_not_use_default_factory() external {
+        // Deploy the contract so we can get the correct deployed code
+        SphinxForkCheck forkCheck = new SphinxForkCheck();
+        address expectedAddress = vm.computeCreate2Address(
+            0,
+            keccak256(type(SphinxForkCheck).creationCode),
+            DETERMINISTIC_DEPLOYMENT_PROXY
+        );
+
+        Vm.AccountAccess[] memory accountAccesses = new Vm.AccountAccess[](2);
+        accountAccesses[0] = VmSafe.AccountAccess({
+            chainInfo: VmSafe.ChainInfo(0, 1),
+            kind: VmSafe.AccountAccessKind.Call,
+            account: address(0x11),
+            accessor: address(this),
+            initialized: false,
+            oldBalance: 0,
+            newBalance: 0,
+            deployedCode: new bytes(0),
+            value: 0,
+            data: type(SphinxForkCheck).creationCode,
+            reverted: false,
+            storageAccesses: new Vm.StorageAccess[](0)
+        });
+        accountAccesses[1] = VmSafe.AccountAccess({
+            chainInfo: VmSafe.ChainInfo(0, 1),
+            kind: VmSafe.AccountAccessKind.Create,
+            account: address(expectedAddress),
+            accessor: DETERMINISTIC_DEPLOYMENT_PROXY,
+            initialized: false,
+            oldBalance: 0,
+            newBalance: 0,
+            deployedCode: address(forkCheck).code,
+            value: 0,
+            data: type(SphinxForkCheck).creationCode,
+            reverted: false,
+            storageAccesses: new Vm.StorageAccess[](0)
+        });
+
+        bool passedCheck = checkAccesses(accountAccesses);
+        assertEq(passedCheck, false);
+    }
+
+    function test_sphinxCheckAccesses_fails_incorrect_deployed_code() external {
```

Comment:
The name of this function indicates that it's supposed to be testing for incorrect deployed code, but it looks like it's actually testing that an incorrect `accountAccesses[1].account` results in `checkAccesses` returning `false`

Reply:
So this is the check this function is designed to test: 
```
// If the deployed code at the calculated address is incorrect, then return false.
// This confirms the deterministic deployment proxy is in fact being used for the
// internal simulation.
if (keccak256(address(expectedAddress).code) != keccak256(type(SphinxForkCheck).runtimeCode)) {
   return false;
}
```

Note that this check *does not* check if the `deployedCode` field is correct in the account access object, but rather it checks that the code at the expected address is correct. This confirms that the deterministic deployment proxy is actually being used by the internal simulation. 

Since we do not actually deploy anything at the expected address in this test, it the deployed code will not be correct and the above check will cause `checkAccesses` to return false.

I do think the name of the test function is misleading though, so I’ll update it.